src/Backups/BackupsWorker.cpp

```diff
@@ -531,13 +520,15 @@ void BackupsWorker::doRestore(
         /// Make a restore coordination.
         if (on_cluster && restore_settings.coordination_zk_path.empty())
         {
-            String root_zk_path = context->getConfigRef().getString("backups.zookeeper_path", "/clickhouse/backups");
-            restore_settings.coordination_zk_path = root_zk_path + "/restore-" + toString(UUIDHelpers::generateV4());
+            restore_settings.coordination_zk_path = root_zk_path + "/restore-" + toString(restore_id);
```

Comment:
`restore_id` is something which could be specified by user. It can be the same for two separate restores if they're not simultaneous. And `coordination_zk_path` must be always unique, that's why it's better to use a random UUID here.

Reply:
updated to use restore_uuid, which is generated.

Reply:
As you mentioned `restore_id` can be specified by user, I believe we use this from restore_settings 
https://github.com/ClickHouse/ClickHouse/blob/0506d9289cfae0cb240b01f5aabf6200c424ba12/src/Backups/BackupsWorker.cpp#L388-L395
In that case if `restore_settings.id` is not empty `restore_id` will be the one specified by user, then we will have 2 restores will same `restore_id` in the map. Seems like this is an issue. Correct me if I am wrong.